src/base/baseAnnotationTool.js

```diff
@@ -42,15 +54,48 @@ export default class extends baseTool {
    * calculated.
    */
   distanceFromPoint (element, data, coords) {
-    console.warn(`distanceFromPoint not implemented for ${this.toolName}`);
+    throw new Error(`Method distanceFromPoint not implemented for ${this.toolName}.`);
   }
 
   /**
-   * Used to redraw the tool's annotation data per render
+   * @abstract Used to redraw the tool's annotation data per render
    *
    * @param {*} evt
    */
   renderToolData (evt) {
-    console.warn(`onImageRendered not implemented for ${this.toolName}`);
+    throw new Error(`renderToolData not implemented for ${this.toolName}.`);
   }
+
+  //===================================================================
+  // Optional Methods - Used to override default behavior.
+  //===================================================================
+
+  /**
+   * Custom callback for when a handle is selected. If overridden, return true.
+   *
+   * @param  {*} evt
+   * @param  {*} handle The selected handle.
+   * @return {boolean} stopImmediatePropagation - True if function consumed the event.
+   */
+  /*
+  handleSelectedCallback (evt, handle) {
+    // Implementation
+    return stopImmediatePropagation;
+  }
+  */
+
+  /**
+   * Custom callback for when a tool is selected. If overridden, return true.
+   *
+   * @param  {*} evt
+   * @param  {*} tool The selected tool.
+   * @return {boolean} stopImmediatePropagation - True if function consumed the event.
+   */
+  /*
+  toolSelectedCallback (evt, tool) {
+    // Implementation
+    return stopImmediatePropagation;
+  }
+  */
```

Comment:
Really like the separation of `abstract` and `optional`. Really easy to follow 👍